### DISCUSSION TITLE
Prevent ESC from closing modal in VueRouter

### DISCUSSION THREAD

**Discussion by fusigabs** [Jenesius/vue-modal#69] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Maybe this exists and I haven't seen it, but it would be nice to be able to pass configuration options when using VueRouter. For instance, I would like to disable closing the modal on `ESC` key press when using VueRouter.

**Describe the solution you'd like**
When I open a modal page using the VueRouter integration, I may have other functionality on that page that requires users to use the `ESC` key. In this case, if the user presses on the `ESC` key, the entire page closes. I'd like a solution where I can disable the `ESC` key when using VueRouter.

**Describe alternatives you've considered**
Maybe something like this to the composable?
```javascript
useModalRouter.init(router, { escClose: false });
```

I know that in theory, we can tap into the `onBeforeModalClose` hook like this:
```javascript
// App.vue

onBeforeModalClose((event) => {
    // prevent modal from closing if the user pressed the `ESC` key
   return false
});
```

However, in this case, it will be nice to add a flag in the `event` to show that the user pressed the `ESC` key. at the moment, the `event` contains only one flag called `background`, which I believe indicates that the user clicked on the background to close the modal. I suggest adding another flag, eg `isESC` or something similar.

**Comment by Jenesius** [comment]
@nfunwigabga I am very grateful to you. 
Thank you for opening this issue. I have already started working on this topic, later I will unsubscribe about the results.

---

I will extend event-close for adding new prop: *esc*
```ts
beforeModalClose(event) {
  event.esc
}
```
Also extend config (Fix error, now is not working): 
```ts
config({
  escClose: false
})
```

I will also think about additional configuration for modal windows based on vue-router

**Comment by Jenesius** [comment, marked as answer]
@nfunwigabga 
Version **1.7.2** include next innovations:
- event-close was extends with *esc* props:
```ts
beforeModalClose(event) {
  if (event.esc) return false; // Modal will not closed by pressing Escape
}
```
- Fix `config({escClose: false})`. Now *true* value will disable handle Escape closing
- Now You can write your own Escape handle:
```ts
import {config, getCurrentModal, closeById} from "../../plugin";
config({
   escClose: false // Disable Escape handle inside library
})

document.addEventListener("keyup", e => {
    if (e.key === "Escape" || e.code === "Escape") {
        const modal = getCurrentModal();
        if (!modal) return; // If no modal was opened
        if (modal.isRoute) return; // New prop. Modal.isRoute will true if modal was opened using useModalRouter
        closeById(modal.id, {esc: true}) // Closing modal by ID. Don't forgot provide esc:true params
    }
})
```
- **getCurrentModal** return Modal if there is opened modal, otherwise null
- **closeById** closing modal by ID. (It's experimental methods, but it using inside closeModal, popModal, modal.close)

**Comment by Jenesius** [comment]
If the current solution is ok, I will move the current issue to a discussion with an answer.

**Comment by fusigabs** [comment]
Hi @Jenesius thanks for working on this so quickly. Yes, this solution works for me. Thanks again.
